src/frontend/Komponenter/Behandling/VedtakOgBeregning/Overgangsstønad/InnvilgeVedtak/InnvilgeVedtak.tsx

```diff
@@ -216,82 +260,99 @@ export const InnvilgeVedtak: React.FC<{
-                        setValideringsFeil={formState.setErrors}
-                        samordningsfradragstype={typeSamordningsfradag}
-                        skalVelgeSamordningstype={skalVelgeSamordningstype}
-                        samordningValideringsfeil={formState.errors.samordningsfradragType}
-                    />
-                </WrapperMarginTop>
-                {behandlingErRedigerbar && (
+                    <Heading spacing size="small" level="5">
+                        Inntekt
+                    </Heading>
+                    {!behandlingErRedigerbar && inntektBegrunnelse.value === '' ? (
+                        <IngenBegrunnelseOppgitt />
+                    ) : (
+                        <EnsligTextArea
+                            value={inntektBegrunnelse.value}
+                            onChange={(event) => {
+                                settIkkePersistertKomponent(VEDTAK_OG_BEREGNING);
+                                inntektBegrunnelse.onChange(event);
+                            }}
+                            label="Begrunnelse for inntektsfastsettelse"
+                            maxLength={0}
+                            erLesevisning={!behandlingErRedigerbar}
+                            feilmelding={formState.errors.inntektBegrunnelse}
+                        />
+                    )}
                     <WrapperMarginTop>
-                        <Knapp type={'standard'} onClick={beregnPerioder} htmlType="button">
-                            Beregn
-                        </Knapp>
+                        <InntektsperiodeValg
+                            inntektsperiodeListe={inntektsperiodeState}
+                            valideringsfeil={formState.errors.inntekter}
+                            setValideringsFeil={formState.setErrors}
+                            samordningsfradragstype={typeSamordningsfradag}
+                            skalVelgeSamordningstype={skalVelgeSamordningstype}
+                            samordningValideringsfeil={formState.errors.samordningsfradragType}
+                        />
                     </WrapperMarginTop>
-                )}
-            </WrapperMarginTop>
-            <WrapperDobbelMarginTop>
-                <Utregningstabell beregnetStønad={beregnetStønad} />
-            </WrapperDobbelMarginTop>
-            {feilmelding && (
-                <AlertStripeFeilPreWrap style={{ marginTop: '2rem' }}>
-                    {feilmelding}
-                </AlertStripeFeilPreWrap>
+                    {behandlingErRedigerbar && (
+                        <WrapperMarginTop>
+                            <Knapp type={'standard'} onClick={beregnPerioder} htmlType="button">
+                                Beregn
+                            </Knapp>
+                        </WrapperMarginTop>
+                    )}
+                </WrapperMarginTop>
+            )}
+            {skalViseVedtaksperiodeOgInntekt && (
```

Comment:
Gjentar denne fire ganger - kunne vi hatt den utenfor én tom fragment-blokk?

Reply:
Tror den første må være for seg selv, da den er på et annet nivå (innenfor `WrapperDobbelMarginTop`), men de to siste trenger kun være én, ja. Fikset i 0fe5cbc.